packages/mobile/src/send/useInviteRewardLimitDialog.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import Dialog from 'src/components/Dialog'
+import { MAX_NUM_INVITES_WITH_REWARDS } from 'src/config'
+import { getAddressFromPhoneNumber } from 'src/identity/contactMapping'
+import {
+  e164NumberToAddressSelector,
+  secureSendPhoneNumberMappingSelector,
+} from 'src/identity/selectors'
+import { Recipient } from 'src/recipients/recipient'
+import { inviteRewardsActiveSelector, invitesWithRewardsSentSelector } from 'src/send/selectors'
+
+const useInviteRewardLimitDialog = (onNavigateSendAmount: (recipient: Recipient) => void) => {
+  const [isVisible, setIsVisible] = useState(false)
+  const [selectedRecipient, setSelectedRecipient] = useState<Recipient | null>(null)
+
+  const invitesWithRewardsSent = useSelector(invitesWithRewardsSentSelector)
+  const inviteRewardsEnabled = useSelector(inviteRewardsActiveSelector)
+  const secureSendPhoneNumberMapping = useSelector(secureSendPhoneNumberMappingSelector)
+  const e164NumberToAddress = useSelector(e164NumberToAddressSelector)
+
+  const { t } = useTranslation()
+
+  const handleClose = () => {
+    setIsVisible(false)
```

Comment:
nit: Shouldn't we also do `setSelectedRecipient(null)`?
Now I'm wondering if we could remove `isVisible` and just use `selectedRecipient`.
Probably it's clearer the way you did it, so feel free to ignore this suggestion.

Reply:
this is a good callout, we can indeed get away with just one of these state variables! i removed the `isVisible` and renamed the recipient to `invitee`, i think it's still pretty readable :D thanks!